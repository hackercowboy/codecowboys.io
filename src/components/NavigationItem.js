import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-scroll';

import './NavigationItem.scss';

function NavigationItem({
  id, target, offset, children,
}) {
  return (
    <Link
      activeClass="active"
      id={id}
      className="navigation-item"
      to={target}
      spy
      offset={offset}
      smooth="true"
      duration={500}
    >
      {children}
    </Link>
  );
}

NavigationItem.propTypes = {
  id: PropTypes.string,
  target: PropTypes.string.isRequired,
  offset: PropTypes.number,
  children: PropTypes.node,
};

NavigationItem.defaultProps = {
  id: undefined,
  offset: 0,
  children: undefined,
};

export default NavigationItem;
