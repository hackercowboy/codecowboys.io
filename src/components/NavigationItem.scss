@import '../styles/colors';

.navigation-item {
  align-items: center;
  cursor: pointer;
  display: flex;
  padding: 5px 10px;
}

.navigation-item {
  &:hover {
    background: $medium-color;
  }
}

.navigation-item.active {
  background-color: $medium-color;
}
