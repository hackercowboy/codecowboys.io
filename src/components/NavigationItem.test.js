import React from 'react';
import { shallow } from 'enzyme';

import NavigationItem from './NavigationItem';

describe('<NavigationItem/>', () => {
  it('should render', () => {
    const wrapper = shallow(<NavigationItem onSetActive={() => true} />);
    expect(wrapper.debug()).toMatchSnapshot();
  });
});
