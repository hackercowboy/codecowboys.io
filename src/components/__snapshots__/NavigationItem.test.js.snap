// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavigationItem/> should render 1`] = `"<Link activeClass=\\"active\\" className=\\"navigation-item\\" to=\\"test\\" spy={true} offset={0} smooth=\\"true\\" duration={500} />"`;
